### PR TITLE
Fixes #17847: Preliminary support for macosx

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/17847

### CODE REVIEW

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
PR updated with a new commit

File: rudder-agent/osx/build.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+base=$(pwd)
+mkdir tmp 
+
+cd ../SOURCES
+# use gnu or system tools as possible (todo do not depend on it since we don't have dependencies management)
+# on OSX 10.15.1
+# - curl 7.64.1
+# - openssl = libressl
+# - no jq, no lmdb, no yaml, no pcre
+build_env="USE_HTTPS=true USE_SYSTEM_OPENSSL=false USE_SYSTEM_FUSION=false USE_SYSTEM_PERL=true USE_SYSTEMD=false USE_SYSTEM_CURL=true USE_SYSTEM_JQ=false USE_SYSTEM_YAML=yaml USE_SYSTEM_XML=true USE_PIE=yes USE_SYSTEM_LMDB=false USE_ACL=false USE_SYSTEM_PCRE=false"
```
Comment (amousset) [inline rudder-agent/osx/build.sh:19]:
```suggestion
build_env="USE_HTTPS=true USE_SYSTEM_OPENSSL=false USE_SYSTEM_FUSION=false USE_SYSTEM_PERL=true USE_SYSTEMD=false USE_SYSTEM_CURL=true USE_SYSTEM_JQ=false USE_SYSTEM_YAML=false USE_SYSTEM_XML=true USE_PIE=yes USE_SYSTEM_LMDB=false USE_ACL=false USE_SYSTEM_PCRE=false"

maybe
```

Comment (peckpeck) [reply rudder-agent/osx/build.sh:19]: indeed

**Comment by peckpeck** [comment]
PR updated with a new commit

**Comment by peckpeck** [comment]
OK, squash merging this PR

**Pull request merged**